src/looprestoration.rs

```diff
@@ -58,93 +59,127 @@ pub struct Dav1dLoopRestorationDSPContext {
     pub sgr: [looprestorationfilter_fn; 3],
 }
 
-#[cfg(feature = "asm")]
-macro_rules! decl_looprestorationfilter_fns {
-    ( $( fn $name:ident, )* ) => {
-        extern "C" {
-            $(
-                // TODO(randomPoison): Temporarily pub until init fns are deduplicated.
-                pub(crate) fn $name(
-                    dst: *mut pixel,
-                    dst_stride: ptrdiff_t,
-                    left: const_left_pixel_row,
-                    lpf: *const pixel,
-                    w: libc::c_int,
-                    h: libc::c_int,
-                    params: *const LooprestorationParams,
-                    edges: LrEdgeFlags,
-                    bitdepth_max: libc::c_int,
-                );
-            )*
+/// Generates a generic wrapper function that delegates to the appropriate
+/// bitdepth-specific extern function.
+macro_rules! decl_looprestorationfilter_bd_fn {
+    ($name:ident, $suffix:ident) => {
+        paste::paste! {
+            #[cfg(feature = "asm")]
+            unsafe extern "C" fn [<$name $suffix>]<BD: BitDepth>(
```

Comment:
We need to check the perf of this extra call.  The asm function won't be inlined unless we have cross language LTO for asm (not sure if we can; it'd be great if we can, though) or something like BOLT probably.  We should be able to do this without this extra call, though.  When I'm back I can open a PR for a way to do this.